engine/src/image.cpp

```diff
@@ -916,6 +917,17 @@ Exec_stat MCImage::getprop(uint4 parid, Properties which, MCExecPoint& ep, Boole
             else
                 ep . clear();
         break;
+        // MERG-2014-09-18: [[ ImageMetadata ]] return the image metadata as an array
+    case P_METADATA:
+        if (m_rep != nil)
+        {
+            MCImageMetadata t_metadata;
+            (static_cast<MCLoadableImageRep *>(m_rep))->GetMetadata(t_metadata);
```

Comment:
This cast is dangerous, as the underlying type of m_rep can easily be something other than MCLoadableImageRep